velox/dwio/parquet/tests/ParquetTpchTestBase.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/parquet/tests/ParquetTpchTestBase.h"
+
+namespace facebook::velox::exec::test {
+
+void ParquetTpchTestBase::SetUpTestSuite() {
+  if (duckDb_ == nullptr) {
+    duckDb_ = std::make_shared<DuckDbQueryRunner>();
+  }
+  tempDirectory_ = TempDirectoryPath::create();
+
+  functions::prestosql::registerAllScalarFunctions();
+  aggregate::prestosql::registerAllAggregateFunctions();
+
+  parse::registerTypeResolver();
+  filesystems::registerLocalFileSystem();
+  velox::parquet::registerParquetReaderFactory();
+
+  auto hiveConnector =
+      connector::getConnectorFactory(
+          connector::hive::HiveConnectorFactory::kHiveConnectorName)
+          ->newConnector(kHiveConnectorId, nullptr);
+  connector::registerConnector(hiveConnector);
+
+  std::shared_ptr<memory::MemoryPool> rootPool{
+      memory::defaultMemoryManager().addRootPool()};
+  std::shared_ptr<memory::MemoryPool> pool{rootPool->addLeafChild("leaf")};
+  auto tpchConnector =
+      connector::getConnectorFactory(
+          connector::tpch::TpchConnectorFactory::kTpchConnectorName)
+          ->newConnector(kTpchConnectorId, nullptr);
+  connector::registerConnector(tpchConnector);
+  saveTpchTablesAsParquet(pool.get());
+  tpchBuilder_.initialize(tempDirectory_->path);
```

Comment:
If we move this logic back to Setup(), we can keep tpchBuilder_ variable non-static.